### PR TITLE
Jobs

### PR DESCRIPTION
**Pull request by luciafalcinelli**

### CODE REVIEW

**Review by dave-mills — CHANGES_REQUESTED** [review]
Hi Lucia, 

Before we merge this into dev and start testing, there are a few things I think we should address - mainly on explaining what's going on to users with the data maps. (This will double as explaining to Ciara and Carlos!)

When you create data maps in the system, I recall you had decided to modify the DataMapController so that you didn't have to match the form variable name exactly to the database variable name. I see that change on line 125 of the datamap controller. 

Can you update the DataMapCrudController - firstly so that it doesn't use the text from the Soils platform, and second so it's clear that the 'name' references the xlsform variable name and the 'label' is actually the sql field name. 

I also see some other references to the soils platform that we don't need. The relationship between the xlsforms and the datamaps is no longer 1 - many - it's actually reversed! (As we have many data maps for a single form). But for now we can just remove the relationship as it's not relevant atm. 

Cheers,
Dave,

File: database/migrations/2020_09_02_084729_create_new_municipio_table.php [hunk 1]
```diff
@@ -28,6 +28,6 @@ public function up()
      */
     public function down()
     {
-        Schema::dropIfExists('municipio');
+        Schema::dropIfExists('new_municipio');
```
[the remark this answers is not in the document]
Comment (luciafalcinelli) [reply database/migrations/2020_09_02_084729_create_new_municipio_table.php:31]: Right, I think it was generated automatically

File: app/Models/Region.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    | FUNCTIONS
+    |--------------------------------------------------------------------------
+    */
+
+    /*
+    |--------------------------------------------------------------------------
+    | RELATIONS
+    |--------------------------------------------------------------------------
+    */
+
```
[the remark this answers is not in the document]
Comment (luciafalcinelli) [reply app/Models/Region.php:37]: The regions are at the top levels for the locations, this means they are before of the departamentos. I added the relationship for all the locations.

File: app/Jobs/GetDataFromKobo.php [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+                //Update or create the record for all the locations 
+
+                if($newSubmission['region'] == 999){
+
+                    $region = Region::updateOrCreate([
+                        'name' => $newSubmission['otra_region']
+                    ]);
+                   
+                    //update the region id value
+                    $newSubmission['region'] = $departamento->id;
```
[the remark this answers is not in the document]
Comment (luciafalcinelli) [reply app/Jobs/GetDataFromKobo.php]: Right, it should be $region->id. I fixed this bug in the last pull.

File: app/Models/Region.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (dave-mills) [reply app/Models/Region.php:37]: Cool - thanks!

File: database/migrations/2020_09_02_084729_create_new_municipio_table.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (dave-mills) [reply database/migrations/2020_09_02_084729_create_new_municipio_table.php:31]: Yeah, it guesses the table name from the migration name. Can you fix this? (It's not yet fixed in the latest commit I see)

[the remark this answers is not in the document]
Comment (luciafalcinelli) [reply database/migrations/2020_09_02_084729_create_new_municipio_table.php]: Sorry, I thought that you did. I fixed now

File: app/Http/Controllers/Admin/DataMapCrudController.php [hunk 4]
```diff
@@ -108,8 +108,8 @@ protected function setupCreateOperation()
                         'wrapper' => ['class' => 'form-group col-md-4'],
                     ],
                     [
-                        'name' => 'label',
-                        'label' => 'Label',
+                        'name' => 'column_name',
```
Comment (dave-mills) [inline app/Http/Controllers/Admin/DataMapCrudController.php:117]: This really isn't enough for someone to know what should go here. The crud controller has references to the soils project with information that is not relevant here; the 'model' variable should not ask about a single data table that the form will populate, as datamaps are per-module, not per-form. 

See my latest pull request for a revised version with detailed labels.

**Review by dave-mills — APPROVED** [review]

**Pull request merged**